docs/developer/branching-and-workflow.md

```diff
@@ -0,0 +1,31 @@
+# Branching and Workflow
+
+NKG is an open source and public repository; our goal is for a spartan and easily manageable set of branches. Ideally,
+our branches are created minimally; there will be a main branch and a handful of release and other long-term feature
+branches.
+
+Internal developers and external contributors will follow a fork and merge process. Each contributor should fork the
+repo to their own space; branch, experiment, develop and merge from their space into NKG’s main branch. NKG branches are
+long-lived, developer branches are ephemeral and unconstrained outside the main repository.
+
+Developer Alice,
+
+- Forks `github.com/nginxinc/nginx-kubernetes-gateway` → `github.com/alice/nginx-kubernetes-gateway`
```

Comment:
Since Alice seem to be a real GitHub user, let's use something like `github.com/<alice-user-id>/...` ?